crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -54,28 +57,33 @@ pub struct OnSettlementEventUpdater {
     pub contract: GPv2Settlement,
     pub native_token: H160,
     pub db: Postgres,
-    pub current_block: CurrentBlockStream,
 }
 
 impl OnSettlementEventUpdater {
-    pub async fn run_forever(self) -> ! {
+    pub async fn run_forever(self, block_stream: CurrentBlockStream) -> ! {
+        let mut current_block = *block_stream.borrow();
+        let mut block_stream = into_stream(block_stream);
         loop {
-            match self.update().await {
-                Ok(true) => (),
-                Ok(false) => tokio::time::sleep(Duration::from_secs(10)).await,
+            match self.update(current_block.number).await {
+                Ok(performed_update) => {
+                    tracing::debug!(
+                        ?performed_update,
+                        block = current_block.number,
+                        "on settlement event updater ran"
+                    );
+                }
                 Err(err) => {
                     tracing::error!(?err, "on settlement event update task failed");
-                    tokio::time::sleep(Duration::from_secs(10)).await;
                 }
             }
+            current_block = block_stream.next().await.expect("blockchains never end");
```

Comment:
We can make it so that update processes all events in a single call. This way we only wait 1 block (which is roughly equivalent to 10s) if processing failed or all events have been processed.

Reply:
I assume there was a reason to do it 1 event at a time but we can probably do it.
However, I would keep the waiting logic as we had them before this PR and only introduce it together with the change you suggested.
That way this PR is just a refactor with very little potential for regressions.

Reply:
I did not implement the original code but probably the reasoning was that 1 by 1 update is better in cases when you need to update 1000 txs at once (let's say we did a manual fix of db and are missing last 2000 blocks) but then the whole update can fails because of 1 bad tx and it's stuck there.